api/v1/databaseengine_types.go

```diff
@@ -39,13 +39,15 @@ type (
 
 // DatabaseEngineSpec is a spec for a database engine.
 type DatabaseEngineSpec struct {
-	Type EngineType `json:"type"`
+	Type            EngineType `json:"type"`
+	EnabledVersions Versions   `json:"enabledVersions,omitempty"`
 }
 
 // DatabaseEngineStatus defines the observed state of DatabaseEngine.
 type DatabaseEngineStatus struct {
-	State   EngineState `json:"status,omitempty"`
-	Version string      `json:"version,omitempty"`
+	State             EngineState `json:"status,omitempty"`
+	Version           string      `json:"version,omitempty"`
```

Comment:
Maybe we should rename this to `OperatorVersion` to avoid confusion with the other versions below. What do you think?

Reply:
What's the point of keeping backward compatibility if we're just weeks away of renaming the project to everest and refactoring the entire API?

Reply:
Yeah. Sure